Sofa/Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/TriangularFEMForceFieldOptim.inl

```diff
@@ -557,6 +565,38 @@ void TriangularFEMForceFieldOptim<DataTypes>::getTrianglePrincipalStress(Index i
     }
 }
 
+
+template<class DataTypes>
+void TriangularFEMForceFieldOptim<DataTypes>::computePrincipalStress()
+{
+    const VecElement& triangles = m_topology->getTriangles();
+
+    sofa::helper::ReadAccessor< core::objectmodel::Data< VecTriangleState > > triStates = d_triangleState;
+    sofa::helper::WriteAccessor< core::objectmodel::Data< VecTriangleInfo > > triInfos = d_triangleInfo;
+
+    Real minStress = 0;
+    Real maxStress = 0;
+
+    for (unsigned int i = 0; i < triangles.size(); i++)
+    {
+        TriangleInfo& triInfo = triInfos[i];
+
+        getTrianglePrincipalStress(i, triInfo.stress, triInfo.stressVector, triInfo.stress2, triInfo.stressVector2);
+
+        if (triInfo.stress < minStress) minStress = triInfo.stress;
+        if (triInfo.stress > maxStress) maxStress = triInfo.stress;
+        if (triInfo.stress2 < minStress) minStress = triInfo.stress2;
+        if (triInfo.stress2 > maxStress) maxStress = triInfo.stress2;
+    }
+
+    maxStress = std::max(-minStress, maxStress);
```

Comment:
Is it possible that both `minStress` and `maxStress` are negative ? In that case, you end up with `maxStress` being positive, far from the actual values.

Reply:
good point... I don't know, I moved that code. I will change it by using max(fabs(min), fabs(max)) instead.